src/views/AddTimer.tsx

```diff
@@ -0,0 +1,183 @@
+// AddTimer.tsx
+
+import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { useState } from 'react';
+import type React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import Button from '../components/button/Button';
+import { useTimerContext } from '../context/TimerContext';
+
+export type TimerType = 'stopwatch' | 'countdown' | 'xy' | 'tabata';
+
+const AddTimer = () => {
+    const [timerType, setTimerType] = useState<TimerType>('stopwatch');
+    const [minutes, setMinutes] = useState(0);
+    const [seconds, setSeconds] = useState(10);
+    const [roundTimeMinutes, setRoundTimeMinutes] = useState(1);
+    const [roundTimeSeconds, setRoundTimeSeconds] = useState(0);
+    const [workTimeMinutes, setWorkTimeMinutes] = useState(0);
+    const [workTimeSeconds, setWorkTimeSeconds] = useState(20);
+    const [restTimeMinutes, setRestTimeMinutes] = useState(0);
+    const [restTimeSeconds, setRestTimeSeconds] = useState(10);
+    const [rounds, setRounds] = useState(1);
+    const [name, setName] = useState('');
+    const [error, setError] = useState('');
+    const { state, dispatch } = useTimerContext();
+    const navigate = useNavigate();
+
+    const handleAddTimer = () => {
+        const duration = minutes * 60 + seconds;
+        const roundTime = roundTimeMinutes * 60 + roundTimeSeconds;
+        const workTime = workTimeMinutes * 60 + workTimeSeconds;
+        const restTime = restTimeMinutes * 60 + restTimeSeconds;
+
+        if (timerType === 'stopwatch' || timerType === 'countdown') {
+            if (duration <= 0) {
+                setError('Duration must be greater than 0.');
+                return;
+            }
+        } else if (timerType === 'xy') {
+            if (roundTime <= 0 || rounds <= 0) {
+                setError('Round time and number of rounds must be greater than 0.');
+                return;
+            }
+        } else if (timerType === 'tabata') {
+            if (workTime <= 0 || restTime <= 0 || rounds <= 0) {
+                setError('Work time, rest time, and number of rounds must be greater than 0.');
+                return;
+            }
+        }
+        setError('');
+
+        const nextNumber = state.timers.length + 1;
+        const defaultName = `Timer ${nextNumber}`;
+
+        const newTimer = {
+            id: uuidv4(),
+            type: timerType,
+            duration,
+            roundTime,
+            workTime,
+            restTime,
+            rounds,
+            name: name.trim() || defaultName,
+            state: 'not running' as const,
+            addedAt: Date.now(),
+            currentRound: 1,
+        };
+
+        dispatch({ type: 'ADD_TIMER', payload: newTimer });
+        dispatch({ type: 'RESET_TIMER_STATE' });
+        navigate('/');
```

Comment:
I would recommend not navigating back to the home route - it becomes inconvenient to add multiple timers